igvc_gazebo/nodes/magnetometer/Magnetometer.cpp

```diff
@@ -0,0 +1,36 @@
+#include "Magnetometer.h"
+
+#include <ros/ros.h>
+#include <geometry_msgs/Vector3Stamped.h>
+#include <sensor_msgs/MagneticField.h>
+
+Magnetometer::Magnetometer()
+{
+	ros::NodeHandle nh;
+    ros::NodeHandle pnh("~");
+    std::string sub_topic = pnh.param("mag_sub_topic", std::string("/magnetometer/vector"));
+    std::string pub_topic = pnh.param("mag_pub_topic", std::string("/magnetometer_mag"));
+    g_mag_field_covar = pnh.param("mag_field_variance", 1e-6);
+    g_mag_field_pub = nh.advertise<sensor_msgs::MagneticField>(pub_topic, 10);
```

Comment:
Here you are still using the `g_` prefix, while the member variable does not have it. Remove the `g_` prefix.